### ISSUE TITLE
[Guacamole] Duo Web SDK v4 - Duo Universal Prompt upgrade issue

### ISSUE THREAD

**Issue by frhodes-aam** [bitnami/vms#1460]
### Platform

AWS

### bndiagnostic ID [know more about bndiagnostic ID](https://docs.bitnami.com/general/how-to/understand-bndiagnostic/)

fbf93fdb-a78c-da16-43e3-56b31e0c6301

### bndiagnostic output

\\
The bndiagnostic tool has finished searching for errors and has found some
issues that might be related to yours. The output will be shown on the next
page:

Press [Enter] to continue:

===== Begin of bndiagnostic tool output =====

    ? Resources: Found possible issues
    ? Connectivity: Found possible issues
    ✓ Processes: No issues found
    ✓ Apache: No issues found

[Resources]

Your instance has little available RAM memory.
```
 total used free shared buff/cache available Mem: 1953 897 107 34 948 865 Swap:
0 0 0
```
You could try to increase your instance's memory. Please check your cloud
provider's documentation for more information.

You can also enable swap memory to improve performance.

Press [Enter] to continue:
https://docs.bitnami.com/installer/faq/linux-faq/administration/increase-m
emory-linux/

[Connectivity]

Server ports 22, 80 and/or 443 are not publicly accessible. Please check the
following guide to open server ports for remote access:

https://docs.bitnami.com/general/faq/administration/use-firewall/

===== End of bndiagnostic tool output =====

### bndiagnostic was not useful. Could you please tell us why?

Not related to my issue

### Describe your issue as much as you can

There is this upgrade done by DUO which changes the way DUO prompt present itself when trying to authenticate, below is a link to the document.

[https://duo.com/docs/duoweb](url)

I am not sure what to do to get Bitnami Guacamole to work with the Universal prompt, I created a new guacamole instance from the latest image and created a new DUO application for it but still it presented the old prompt. This is a problem for us as we need to get this to work before the end of this week.

Will appreciate a quick response.

Thank you.

Regards

Fego Rhodes

**Comment by jotamartos** [comment]
>I am not sure what to do to get Bitnami Guacamole to work with the Universal prompt, I created a new guacamole instance from the latest image and created a new DUO application for it but still it presented the old prompt. This is a problem for us as we need to get this to work before the end of this week.

Thank you for using Bitnami. In this case, you are customizing the app so I suggest you ask the DUO developers to get more information about how to install and configure it. Please remember that the Guacamole files are inside the /opt/bitnami/guacamole directory. You can find there all the binaries and configuration files.

**Comment by frhodes-aam** [comment]
> > I am not sure what to do to get Bitnami Guacamole to work with the Universal prompt, I created a new guacamole instance from the latest image and created a new DUO application for it but still it presented the old prompt. This is a problem for us as we need to get this to work before the end of this week.
> 
> Thank you for using Bitnami. In this case, you are customizing the app so I suggest you ask the DUO developers to get more information about how to install and configure it. Please remember that the Guacamole files are inside the /opt/bitnami/guacamole directory. You can find there all the binaries and configuration files.

Hi Jotamartos

Thanks for your response.

Now the issue is that this is not me wanting to customize the bitnami guacamole offering, DUO has moved from using the old integration method to the new one and Bitnami will need to ask their developers to incorporate the changes into their application else after some time no one using bitnami Guacamole can use the DUO Integration as it is ending support in a few days but the end of life is near too.

So Bitnami will need to take action on it and not me modifying their application to suit my needs, it is a general requirement by DUO. Moreover I am not a developer, I pay to use the Bitnami Guacamole product and the developers should do their job.

Thank you.

**Comment by jotamartos** [comment]
>DUO has moved from using the old prompt method to a new one and Bitnami will need to ask their developers to incorporate the changes into their application, else after some time no one using bitnami Guacamole can use the DUO product for 2FA as it is ending support in a few days, but the end of life is near too.

**Comment by mcbsys** [comment]
Updating the Duo plugin for Guacamole is something that Apache Guacamole must fix. It looks like they plan to release a fix in 1.6.0:

https://issues.apache.org/jira/browse/GUACAMOLE-1289?jql=text%20~%20%22duo%20universal%22

According to their releases page. as of today (23 March 2025), the current release is 1.5.5 (on 05 April 2024):

https://guacamole.apache.org/releases/
